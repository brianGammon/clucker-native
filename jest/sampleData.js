const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
