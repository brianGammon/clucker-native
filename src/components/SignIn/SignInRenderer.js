import * as React from 'react';
import { FieldGroup, FieldControl } from 'react-reactive-form';
import { Text, Button, Form } from 'native-base';
import FormInput from '../FormInput';
import AuthContainer from '../Auth/AuthContainer';
import AuthExtraLink from '../Auth/AuthExtraLink';

type Props = {
  navigation: any,
  error: string,
  loginForm: any,
  handleSubmit: any,
};

const SignInRenderer = ({
  navigation,
  error,
  loginForm,
  handleSubmit,
}: Props) => (
  <AuthContainer
    title="Sign In"
    error={error}
    extraLinks={[
      <AuthExtraLink
        key={0}
        message="No account yet?"
        buttonText="Sign Up"
        handlePress={() => navigation.replace('SignUp')}
      />,
      <AuthExtraLink
        key={1}
        message="Forgot your password?"
        buttonText="Reset"
        handlePress={() => navigation.navigate('ResetPassword')}
      />,
    ]}
  >
    <FieldGroup
      control={loginForm}
      render={({ invalid }) => (
        <Form>
          <FieldControl
            name="email"
            render={FormInput}
            meta={{
              label: 'Email Addess',
              autoCapitalize: 'none',
              keyboardType: 'email-address',
            }}
          />

          <FieldControl
            name="password"
            render={FormInput}
            meta={{
              label: 'Password',
              autoCapitalize: 'none',
              secureTextEntry: true,
            }}
          />
          <Button
            style={{ marginTop: 10 }}
            block
            onPress={handleSubmit}
            disabled={invalid}
          >
            <Text>Submit</Text>
          </Button>
        </Form>
      )}
    />
  </AuthContainer>
);

export default SignInRenderer;
