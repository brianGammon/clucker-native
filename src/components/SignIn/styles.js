import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  formContainer: {
    width: '100%',
    alignItems: 'center',
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    marginTop: 8,
  },
  text: {
    fontSize: 30,
  },
  error: {
    color: 'red',
  },
});
