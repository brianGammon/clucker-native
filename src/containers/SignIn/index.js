import React from 'react';
import {
  Text, View, Button, AsyncStorage,
} from 'react-native';

export interface Props {
  navigation: any;
}

export default class SignIn extends React.Component<Props> {
  static navigationOptions = {
    title: 'Sign In',
  };

  constructor() {
    super();
    this.handleSignIn = this.handleSignIn.bind(this);
  }

  async handleSignIn() {
    const { navigation } = this.props;
    await AsyncStorage.setItem('userToken', 'TestUser');
    navigation.navigate('App');
  }

  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>Sign In Screen</Text>
        <Button onPress={this.handleSignIn} title="Be Signed In" />
        <Button onPress={() => navigation.navigate('ResetPassword')} title="Reset Password" />
        <Button onPress={() => navigation.replace('SignUp')} title="Sign Up" />
        <Button onPress={() => navigation.navigate('ResetPassword')} title="Reset Password" />
      </View>
    );
  }
}
