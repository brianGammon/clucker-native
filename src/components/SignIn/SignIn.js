import * as React from 'react';
import { connect } from 'react-redux';
import { FormBuilder, Validators } from 'react-reactive-form';
import SignInRenderer from './SignInRenderer';
import { signInRequested } from '../../redux/actions';
import { actionTypes } from '../../redux/constants';

type Props = {
  navigation: any,
  error: string,
  signIn: (email: string, password: string) => void,
  clearError: () => void,
};

class SignIn extends React.Component<Props> {
  loginForm = FormBuilder.group({
    email: ['', [Validators.email, Validators.required]],
    password: ['', Validators.required],
  });

  componentWillUnmount() {
    const { error, clearError } = this.props;
    if (error) {
      clearError();
    }
  }

  handleSubmit = () => {
    const { email, password } = this.loginForm.value;
    const { signIn } = this.props;
    signIn(email, password);
  };

  render() {
    const { navigation, error } = this.props;
    return (
      <SignInRenderer
        navigation={navigation}
        error={error}
        loginForm={this.loginForm}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

const mapStateToProps = ({ auth: { errors } }) => ({
  error: errors.signIn,
});

const mapDispatchToProps = dispatch => ({
  signIn: (email, password) => dispatch(signInRequested(email, password)),
  clearError: () => dispatch({ type: actionTypes.CLEAR_AUTH_ERROR }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignIn);
