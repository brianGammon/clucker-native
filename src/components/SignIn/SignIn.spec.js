import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/no-flocks.json';
import SignIn from '.';

describe('SignIn component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);
  test('Should render', () => {
    const wrapper = shallow(<SignIn store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
