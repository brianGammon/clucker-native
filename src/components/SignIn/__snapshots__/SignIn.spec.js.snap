// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component: Should render 1`] = `
<SignInRenderer
  email=""
  error=""
  handleChangeText={[Function]}
  handleSignIn={[Function]}
  password=""
/>
`;
